CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,194 @@
+* Sustained disruption of community spaces or events. This includes brining
+  weaponry into community spaces; see below for a more precise definition.
+* Intentional violation of local laws or regulations at in-person events.
+* Advocating for, encouraging, or assisting in any of the above behaviour.
+
+'Weapons' include, but are not limited to, guns, explosives (including 
+fireworks without prior approval by a community organizer), and large knives 
+such as those used for hunting or display, as well as any other item used for 
+the purpose of causing injury and harm to others. 
+
+## Consequences of unacceptable behaviour
+
+Unacceptable behaviour from any community member, including anyone with
+decision-making authority, will not be tolerated. Anyone asked to stop
+unacceptable behaviour is expected to comply immediately.
+
+If a community member engages in unacceptable behaviour, the community
+organizers may take any action they deem appropriate, subject to correction 
+and reversal by the lead organizer. These include, but are not limited to:
+
+* A verbal warning (either in private or in public).
+* Eviction from an event with no refund (if applicable), with conditional 
+  right of return or without right of return.
+* A temporary ban of any length, with or without right of appeal, from any or
+  all community spaces.
+* A permanent ban, with or without right of appeal, from any or all community
+  spaces.
+* Temporary revocation of community organizer status for any length of time, 
+  with or without right of appeal.
+* Permanent revocation of community organizer status with or without right of 
+  appeal.
+* Removal of contributions to Pext from any Pext-associated repositories, with
+  or without right of appeal.
+* Notification of law enforcement, or other appropriate authorities, along with
+  any evidence of wrongdoing provided to the community organizers.
+
+## Reporting guidelines
+
+If you are subject to, or witness, unacceptable behaviour, or have any other
+concerns, please notify a community organizer as soon as possible. The current
+list of community organizers is:
+
+* Sylvia van Os <sylvia@hackerchick.me> (lead organizer)
+
+When reporting, please include the following information, to the best of your
+knowledge and ability:
+
+* When and where the unacceptable behaviour occurred.
+* Who engaged in unacceptable behaviour.
+* Which unacceptable behaviour(s) happened.
+* Any evidence for your report (such as screencaps, logs, etc)
+* Who (besides yourself) was affected by this behaviour.
+* Whether you want any additional actions taken by the community organizers to 
+  ensure that your safety and well-being remain intact.
+
+Additionally, community organizers will assist local authorities and connect you
+with help or providing a safe environment inasfar as they are able. In the 
+context of in-person events, organizers will also provide escorts as desired by 
+the person experiencing distress.
```

Comment:
"inasfar" seems to not be a real word, I think you mean "insofar".

I also don't see the same "inasfar as they are able" for the escort part.

How about adding it to the last bullet point?

"Whether you want any additional actions taken by the community organizers to ensure that your safety and well-being remain intact. For example, if so desired and to the best of their ability, community organizers will assist local authorities, connect you with help, provide a safe environment or escort you in case of an in-person event."

Unless I missed why it's written as "additionally"?

Reply:
Yes, I did; good catch, will amend. With regard to providing escorts at in-person events, I don't think this is something that should be up for a competency-based opt-out. With legal stuff, this is definitely somewhere we could (and should!) plead incompetence where appropriate, but setting up someone to accompany someone in distress isn't really something we can claim to be unable to do - if we fail there, we've failed basic organization.

The 'additionally' is from the original source. I agree that it's misplaced, and will remove it.

Reply:
While I get your point, I have a bit of an allergy for promising anything I'm not 100% sure I can do right. For example, if there will be a Pext event, it'll most likely only be me on the team. As soon as you have two people in distress, that can already become a problem, especially if those two don't get along well.

Personally, I find it worse to be promised with certainty something can be provided and then not get, then being told they will try their best so I can be mentally prepared for it possibly not going 100% perfectly.

Reply:
If there will be a Pext event, it'll be:

1. So far into the future that who will be on the org team is very much up for debate; and
2. Under the auspices of a much larger conference, with its own Code of Conduct and its own org team.

Pext's community would have to grow by _at least_ an order of magnitude from where it currently is for an in-person event to even be a thing we can consider, with appropriate up-scaling of the org team.

Thus, I understand your concerns, but I still don't think this is something you need to worry about. My concern is that over-softening the text makes it come across as limp and ineffectual, which isn't something you want a Code of Conduct to be. I'll make the changes if you really insist, but I find them misplaced at best and actually detrimental to the document at worst, for no good reason.

Reply:
Well, here you are talking about the future. Why couldn't I just host a small Pext meetup at some Linux event with a few people? The Code of Conduct would already apply then. I don't think it's good to only look at the future and just say "Pext is too small for an event now, so it's not a problem the rules are broken for the current situation".

Reply:
@VenomousDingo What do you think about this whole thing?

Reply:
Would said Linux event have its own Code of Conduct, with its own org team that you could ask for help? If so, you could always ask for assistance from them if this kind of issue comes up, as it's also their responsibility. I understand your reservations, but I honestly don't share them, precisely because the Pext community is currently far too small for its org team to be _unilaterally_ responsible for organizing and running an in-person event for it.

I really feel that softening this language would reduce confidence in the Code of Conduct for Pext. I understood the reservations (and made changes as appropriate) with matters of legal responsibility, because that's a serious issue on which pleading incompetence is not only easier for the organizers, but is actually the right thing to do. In this case, however, I really don't believe it's right to plead incompetence, precisely because I don't foresee any unilaterally-run Pext events until the community becomes so much larger than it currently is that it would necessitate considerable expansion of the org team. Until that point, you'd be part of a parent event anyway, whose team should (and would!) be available to assist with something like this, as that would be part of their jurisdiction also.

Reply:
I just personally flat out disagree with promising anything. Using the words "may" or "can" should suffice but I don't like the usage of "will" in this at all. This can set us up for a legal mess in the future.

Reply:
@kozross, you said:
>precisely because I don't foresee any unilaterally-run Pext events until the community becomes so much larger than it currently is that it would necessitate considerable expansion of the org team

The @AsteroidOS community sometimes has small meetups "whereever the developer ends up being" (see https://twitter.com/AsteroidOS/status/897508413692923904). I really do want to organize small events like that, even while Pext is still fairly small. Which would already be in-person events.

Like, I would maybe consider making the language more certain when we have bigger events and a bigger team, but I want a Code of Conduct that will work now already too for the small events that may be hosted at some point.